internal/controller/postgrescluster/postgres.go

```diff
@@ -628,6 +635,71 @@ func (r *Reconciler) reconcilePostgresDataVolume(
 	return pvc, err
 }
 
+// setVolumeSize compares the potential sizes from the instance spec, status
+// and limit and sets the appropriate current value.
+func (r *Reconciler) setVolumeSize(ctx context.Context, cluster *v1beta1.PostgresCluster,
+	pvc *corev1.PersistentVolumeClaim, instanceSpecName string) {
+	log := logging.FromContext(ctx)
+
+	// Capture the largest pgData volume size currently defined for a given instance set.
+	volumeRequestSize := pvc.Spec.Resources.Requests.Storage() //.Value()
```

Comment:
Maybe do the same for the limit for readability's sake? (I thought I had this in my review before, but it was lost somehow)
```suggestion
	volumeRequestSize := pvc.Spec.Resources.Requests.Storage()
	volumeLimitSize := pvc.Spec.Resources.Limits.Storage()
```

Reply:
I'll definitely get rid of the obsolete comment, but part of what I'm trying to do is distinguish the values from the spec vs the value that I'm determining (i.e. `volumeRequestSize`). I'm concerned that renaming here will make that distinction less clear. Also, when I'm ultimately setting `pvc.Spec.Resources.Requests`, it's done by reference, so I'm trying to avoid any extra copies if possible.

Reply:
I thought about this a bit more and talked it over with @tony-landreth. I think we can maintain the distinction with carefully named variables. I'll push a change in a few. Please let me know what you think.

Reply:
Looks good